homeassistant/components/cloud/account_link.py

```diff
@@ -72,7 +76,7 @@ async def _get_services(hass: HomeAssistant) -> list[dict[str, Any]]:
             hass.data[DATA_CLOUD]
         )
     except (aiohttp.ClientError, TimeoutError):
-        return []
+        return None
```

Comment:
I think we should raise an exception here instead. It's weird to catch an exception, return None, and then raise an exception.

We could simply raise the `ImplementationUnavailableError` already here.